explorer/src/lib/components/transaction-details/TransactionDetails.svelte

```diff
@@ -53,8 +54,47 @@
   /** @type {boolean} */
   let isPayloadToggled = false;
 
+  /** @type {boolean} */
+  let isMemoDecoded = false;
+
   $: classes = makeClassName(["transaction-details", className]);
 
+  /**
+   * @param {string} hex
+   * @returns {string}
+   */
+  function checkHexString(hex) {
+    if (!isHex(hex)) {
+      return hex;
+    }
+
+    /**
+     * @returns {Uint8Array}
+     */
+    const hexToBytes = () => {
+      const bytes = new Uint8Array(hex.length / 2);
+      for (let i = 0; i < hex.length; i += 2) {
+        bytes[i / 2] = parseInt(hex.slice(i, i + 2), 16);
+      }
+      return bytes;
+    };
+
+    const bytes = hexToBytes();
+    const decoder = new TextDecoder();
+
+    //Check whether the string contains almost all visible Unicode characters up to the end of the BMP
+    /* eslint-disable-next-line no-control-regex */
+    if (/^[\u0000-\u007F\u00A0-\uFFFF]*$/.test(decoder.decode(bytes))) {
```

Comment:
You're decoding twice – once to check the regex and then within the `try`. You could optimize things a bit and only decode once and then store the result.

Something like:

```
  let decodedString;
  try {
    decodedString = decoder.decode(bytes);
  } catch {
    return hexString; // If decoding fails, return the original hex string.
  }
```